tools/@aws-cdk/spec2cdk/lib/cdk/resource-class.ts

```diff
@@ -105,6 +129,22 @@ export class ResourceClass extends ClassType {
       cfnMapping.add(prop.cfnMapping);
     }
 
+    // Build the shared interface
+    for (const identifier of this.decider.primaryIdentifier ?? []) {
+      this.resourceInterface.addProperty({
+        ...identifier,
+        immutable: true,
+      });
+    }
+
+    // Add the arn too, unless it is duplicated in the resourceIdentifier already
+    if (this.decider.arn && this.resourceInterface.properties.every((p) => p.name !== this.decider.arn!.name)) {
```

Comment:
@mrgrain  this fixes the duplicate identifier situation in vpc lattice (i'm 99% sure) with the test case to show it